webui/react/src/hooks/useModal/useJupyterLabModal.tsx

```diff
@@ -116,42 +113,73 @@ const JupyterLabModal: React.FC<JupyterLabModalProps> = (
         slots: fields.slots,
         templateName: fields.template,
       });
+      modalClose();
     }
-    if (onLaunch) onLaunch();
-  }, [ config, fields, launchJupyterLab, onLaunch, showFullConfig ]);
+  }, [ config, fields, launchJupyterLab, showFullConfig, modalClose ]);
 
   const handleConfigChange = useCallback((config: string) => setConfig(config), []);
+  const formContent = useMemo(() => showFullConfig ? (
+    <>
+      <JupyterLabFullConfig
+        config={config}
+        setButtonDisabled={setButtonDisabled}
+        onChange={handleConfigChange}
+      />
+    </>
+  ) : (
+    <>
+      <JupyterLabForm fields={fields} onChange={dispatch} />
+    </>
+  ), [ config, dispatch, fields, handleConfigChange, showFullConfig ]);
+
+  const content = useMemo(() => (
+    <div className={css.modalContent}>
+      {formContent}
+      <div className={css.buttons}>
+        <Button
+          onClick={handleSecondary}>
+          {showFullConfig ? 'Show Simple Config' : 'Show Full Config'}
+        </Button>
+        <Button
+          disabled={buttonDisabled}
+          type="primary"
+          onClick={handleCreateEnvironment}>Launch
+        </Button>
+      </div>
+    </div>
+  ), [ formContent, buttonDisabled, handleCreateEnvironment, handleSecondary, showFullConfig ]);
 
-  return (
-    <Modal
-      footer={(
-        <>
-          <Button
-            onClick={handleSecondary}>
-            {showFullConfig ? 'Show Simple Config' : 'Show Full Config'}
-          </Button>
-          <Button
-            disabled={buttonDisabled}
-            type="primary"
-            onClick={handleCreateEnvironment}>Launch
-          </Button>
-        </>
-      )}
-      title="Launch JupyterLab"
-      visible={visible}
-      width={540}
-      {...props}>
-      {showFullConfig ? (
-        <JupyterLabFullConfig
-          config={config}
-          setButtonDisabled={setButtonDisabled}
-          onChange={handleConfigChange}
-        />
-      ) : (
-        <JupyterLabForm fields={fields} onChange={dispatch} />
-      )}
-    </Modal>
-  );
+  const modalProps: ModalFuncProps = useMemo(() => {
+    return {
+      className: css.noFooter,
+      content: content,
+      title: 'Launch JupyterLab',
+      width: 540,
+    };
+  }, [ content ]);
```

Comment:
**nit suggestion:** slightly shorter but just as easy to read format:
```
const modalProps: ModalFuncProps = useMemo(() => ({
  className: css.noFooter,
  ...
}), [ content ]);
```
cuts down 2 lines